servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -210,8 +219,32 @@ public Completable apply(final int count, final Throwable t) {
             }
 
             Completable applyRetryCallbacks(final Completable completable, final int retryCount, final Throwable t) {
-                return retryCallbacks == null ? completable :
-                        completable.beforeOnComplete(() -> retryCallbacks.beforeRetry(retryCount, requestMetaData, t));
+                Completable result = (retryCallbacks == null ? completable :
+                        completable.beforeOnComplete(() -> retryCallbacks.beforeRetry(retryCount, requestMetaData, t)));
+                if (returnFailedResponses) {
+                    if (t instanceof HttpResponseException &&
+                            ((HttpResponseException) t).metaData() instanceof StreamingHttpResponse) {
+                        StreamingHttpResponse response = (StreamingHttpResponse) ((HttpResponseException) t).metaData();
+                        // If we succeed, we need to drain the response body before we continue. If we fail we want to
+                        // surface the original exception and don't worry about draining since it will be returned to
+                        // the user.
+                        result = result.onErrorMap(backoffError -> ThrowableUtils.addSuppressed(t, backoffError))
+                                // If we get cancelled we also need to drain the message body as there is no guarantee
+                                // we'll ever receive a completion event, error or success.
```

Comment:
Thanks for clarification! Consider adding: and it's legit to do that because subscriber is no longer interested in the response